org.spoofax.terms/src/org/spoofax/terms/io/TermWriter.java

```diff
@@ -0,0 +1,60 @@
+package org.spoofax.terms.io;
+
+import org.spoofax.interpreter.terms.IStrategoTerm;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+
+/**
+ * Interface for term writers.
```

Comment:
I added a description. `SAFWriter` cannot implement this interface because it takes the input term in the constructor. Since @Apanatshka is looking at `SAFWriter` anyway, perhaps there is a way to make it implement `TermWriter` with minimal changes.

Reply:
It's probably possible to modify `SAFWriter` to implement `TermWriter`. It has internal state that would have to be cleared after each call to write a term. Feel free to try that.

Reply:
Implemented `TermWriter` for `SAFWriter`.